File: QUEUE/queue.cpp
#include "queue.h"
using namespace std;

Queue::Queue()
{
    f=r=-1;
}

Queue::~Queue(){}

bool Queue::isEmpty()
{
    return (r==-1);
}

bool Queue::isFull()
{
    return (r>=MAX_QUEUE_SIZE-1);
}

void Queue::enqueue(int val)
{
    if (!isFull())
    {
        r++;
        this->queue_array[r]=val;
    }
    else
    {
        cout<<"Queue overflow."<<endl;
    }
}

int Queue::dequeue()
{
    if (!isEmpty())
    {
        return this->queue_array[++f];
    }
    else
    {
        cout<<"Queue underflow."<<endl;
    }
}

int Queue::get_front()
{
    if (!isEmpty())
    {
        return this->queue_array[++f];
    }
    else
    {
        cout<<"Queue underflow."<<endl;
    }
}

int Queue::get_rear()
{
    if (!isEmpty())
    {
        return this->queue_array[r];
    }
    else
    {
        cout<<"Queue underflow."<<endl;
    }
}

File: Python/queue/queue.py
class Queue():
    def __init__(self, queue_size):
        self.queue_size = queue_size
        self.elements = []

    def enqueue(self, ele):
        if not self.is_full:
            self.elements.append(ele)
        else:
            print("Queue Overflow!")

    def dequeue(self):
        if not self.is_empty:
            return self.elements.pop(0)
        else:
            print("Queue Underflow!")
    
    @property
    def front(self):
        if not self.is_empty:
            return self.elements[0]
        else:
            print("Queue Underflow!")

    @property
    def rear(self):
        if not self.is_empty:
            return self.elements[-1]
        else:
            print("Queue Underflow!")

    @property
    def is_empty(self):
        return len(self.elements) == 0

    @property
    def is_full(self):
        return len(self.elements) == self.queue_size
